fenix/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt

```diff
@@ -865,6 +868,39 @@ class DefaultTabsTrayControllerTest {
         verify { controller.handleNavigateToBrowser() }
     }
 
+    @Test
+    fun `GIVEN a private tab is open and selected with a normal tab also open WHEN the private tab is closed and private home page shown and normal tab is selected from tabs tray THEN normal tab is displayed  `() {
+        val normalTab = TabSessionState(
+            content = ContentState(url = "https://simulate.com", private = false),
+            id = "normalTab",
+        )
+        val privateTab = TabSessionState(
+            content = ContentState(url = "https://mozilla.com", private = true),
+            id = "privateTab",
+        )
+        browserStore = BrowserStore(
+            initialState = BrowserState(
+                tabs = listOf(normalTab, privateTab),
+                selectedTabId = privateTab.id,
+            ),
+        )
+        browsingModeManager.mode = BrowsingMode.Private
+        val controller = spyk(createController())
+
+        try {
+            mockkStatic("mozilla.components.browser.state.selector.SelectorsKt")
+            controller.handleTabSelected(privateTab, null)
+            controller.handleTabDeletion("privateTab")
+
+            browserStore.dispatch(TabListAction.SelectTabAction(normalTab.id)).joinBlocking()
+            controller.handleTabSelected(normalTab, null)
+
+            assertEquals(normalTab.id, browserStore.state.selectedTabId)
```

Comment:
Since the crux of this PR is adding the logic to set the browsing mode, can you add some assertions here to validate its being set properly? There's two calls to `handleTabSelected`, so I'd imagine there'd be an assertion after each call.

Reply:
added the assertion call after the initial `handleTabSelected` and changed the `browserStore` to not have an initial `selectedTabId`

Reply:
I'm only seeing assertions for the selected tab ID. Could you add some for the browsing mode, since that's the logic change in this PR? This is good to go when those are added.

Reply:
ok also changed the test to include checks for the browsing mode